src/index.ts

```diff
@@ -180,12 +189,17 @@ function Miniget(url: string | URL, options: Miniget.Options = {}): Miniget.Stre
         parsed = opts.transform(parsed);
       } catch (err) {
         stream.emit('error', err);
+        // Annotation: this return does not stop execution :thinking:
+        // it only returns the catch-block and continues with Line 196
```

Comment:
i don't think this is true, i ran [this test](https://github.com/fent/node-miniget/blob/master/test/request-test.ts#L390) and logged execution, it stopped here